### PR TITLE
feat: Add Background Job

### PR DESCRIPTION
**Pull request by zhenlei520**
# Description

Add Background Job

### CODE REVIEW

**Comment by doddgu** [comment]
3 Code Smells, and 55.6 Coverage.

File: src/Contrib/Extensions/Masa.Contrib.Extensions.BackgroundJobs/BackgroundJobOptions.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+namespace Masa.Contrib.Extensions.BackgroundJobs;
+
+public class BackgroundJobOptions
+{
+    private int _pollInterval = 5;
+
+    /// <summary>
+    /// Background task rotation interval period
+    /// default: 5s
```
Comment (doddgu) [inline src/Contrib/Extensions/Masa.Contrib.Extensions.BackgroundJobs/BackgroundJobOptions.cs:12]: default? Default?

File: src/Contrib/Extensions/Masa.Contrib.Extensions.BackgroundJobs/BackgroundJobOptions.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+            _batchSize = value;
+        }
+    }
+
+    private int _maxRetryTimes = 30;
+
+    /// <summary>
+    /// maximum number of retries
+    /// Default is 10
```
Comment (doddgu) [inline src/Contrib/Extensions/Masa.Contrib.Extensions.BackgroundJobs/BackgroundJobOptions.cs:45]: default? Default?

File: src/Contrib/Extensions/Masa.Contrib.Extensions.BackgroundJobs/BackgroundJobOptions.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        {
+            MasaArgumentException.ThrowIfLessThan(value, 0, nameof(MaxRetryTimes));
+
+            _maxRetryTimes = value;
+        }
+    }
+
+    private int _firstWaitDuration = 60;
+
+    public int FirstWaitDuration
```
Comment (doddgu) [inline src/Contrib/Extensions/Masa.Contrib.Extensions.BackgroundJobs/BackgroundJobOptions.cs:60]: s? ms? m?

File: src/Contrib/Extensions/Masa.Contrib.Extensions.BackgroundJobs/BackgroundJobOptions.cs [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        {
+            MasaArgumentException.ThrowIfLessThanOrEqual(value, 0, nameof(FirstWaitDuration));
+
+            _firstWaitDuration = value;
+        }
+    }
+
+    private int _waitDuration = 2;
+
+    public int WaitDuration
```
Comment (doddgu) [inline src/Contrib/Extensions/Masa.Contrib.Extensions.BackgroundJobs/BackgroundJobOptions.cs:73]: s? ms? m?

File: src/Contrib/Extensions/Masa.Contrib.Extensions.BackgroundJobs/BackgroundJobProcessor.cs [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+                    job.IsInvalid = true;
+                }
+
+                await backgroundJobStorage.UpdateAsync(job);
+            }
+            catch (Exception ex)
+            {
+                Logger?.LogError(ex, "----- A background job execution is failed. Job: {JobInfo}", job);
+
+                var nextTryTime = NextTryTime(job);
```
Comment (doddgu) [inline src/Contrib/Extensions/Masa.Contrib.Extensions.BackgroundJobs/BackgroundJobProcessor.cs:83]: Same as above except logger message?

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/Contrib/Extensions/Masa.Contrib.Extensions.BackgroundJobs/BackgroundJobProcessor.cs (hunk 5 above)
Comment (zhenlei520) [reply src/Contrib/Extensions/Masa.Contrib.Extensions.BackgroundJobs/BackgroundJobProcessor.cs]: Processed

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**